### ISSUE TITLE
reproduce results

### ISSUE THREAD

**Issue by zouguojian** [liyaguang/DCRNN#78]
Who can reproduce similar results on the METR-LA dataset? My result is worse than this paper.

**Comment by liyaguang** [comment]
Thanks for the interests in the work and the feedback. The following tips may help reproduce the results:
- Use the python packages with versions recommended in the [Requirement section](https://github.com/liyaguang/DCRNN#requirements), ideally with tensorflow version 1.3.0 since I have not tested the training after that version.
- Check possible gradient explosion or adjust the learning rate as recommended in the [training details section](https://github.com/liyaguang/DCRNN#training-details-and-tensorboard-links). _There is a chance of training loss explosion, one temporary workaround is to restart from the last saved model before the explosion, or to decrease the learning rate earlier in the learning rate schedule._
- Here are the related tensorboard links when training the models,  [DCRNN on METR-LA](https://tensorboard.dev/experiment/ijwg04waSOWQ2Pj4mZ3tAg), [DCRNN on PEMS-BAY](https://tensorboard.dev/experiment/QzJtnMfgQJCQ7vc7wNJjxg). You may use them as a reference, e.g., comparing the train/eval loss curves, eval metrics at different time horizons.

Hope this can help, and feel free to update this thread if you have encountered issues.

**Comment by zouguojian** [comment]
Yep, it does not work in the METR-LA dataset. Although I have used the same codes on the server, it is worse than the results you give. Of course, I have read the tips you provide, but it is non-helpful for me. I want to know how others' experimental results. Thank you for your reply.

**Comment by zouguojian** [comment]
![Uploading Screen Shot 2023-03-06 at 12.47.56 PM.png…]()

**Comment by zouguojian** [comment]
<img width="864" alt="Screen Shot 2023-03-06 at 12 47 56 PM" src="https://user-images.githubusercontent.com/32932644/223021903-dd6fb4a6-a86c-46c1-b277-5352494efa55.png">

**Comment by zouguojian** [comment]
> <img alt="Screen Shot 2023-03-06 at 12 47 56 PM" width="864" src="https://user-images.githubusercontent.com/32932644/223021903-dd6fb4a6-a86c-46c1-b277-5352494efa55.png">

it does not work

**Comment by zouguojian** [comment]
> <img alt="Screen Shot 2023-03-06 at 12 47 56 PM" width="864" src="https://user-images.githubusercontent.com/32932644/223021903-dd6fb4a6-a86c-46c1-b277-5352494efa55.png">

Attention, please; horizon 13 is the average result.

**Comment by liyaguang** [comment]
Thank you for providing the log, and spending time debugging this. It seems that there may be some discrepancies, as even a naive baseline approach considering constant traffic could yield significantly better accuracy. Additionally, would you kindly share the version of Tensorflow that you are currently using? It is important to note that the provided code has solely been tested with Tensorflow 1.x, and may not function properly with Tensorflow 2.x. 

Attached is my training log from 2020 with Tensorflow 1.14.0, which I hope will be of assistance: https://gist.github.com/liyaguang/f689b004f6aa04f6767bbadaeb808d47. 

Furthermore, I have conducted another run today, the results of which were quite comparable. In the following days, I plan to allocate time to test it on Tensorflow 2.0.

**Comment by zouguojian** [comment]
> 

Dear Li, it's okay to test codes on the pre-trained model weights you give (i.e., ).  However, I couldn't get the results when I trained the model myself.  The version of TensorFlow is 1.14.

**Comment by zouguojian** [comment]
> Thank you for providing the log, and spending time debugging this. It seems that there may be some discrepancies, as even a naive baseline approach considering constant traffic could yield significantly better accuracy. Additionally, would you kindly share the version of Tensorflow that you are currently using? It is important to note that the provided code has solely been tested with Tensorflow 1.x, and may not function properly with Tensorflow 2.x.
> 
> Attached is my training log from 2020 with Tensorflow 1.14.0, which I hope will be of assistance: https://gist.github.com/liyaguang/f689b004f6aa04f6767bbadaeb808d47.
> 
> Furthermore, I have conducted another run today, the results of which were quite comparable. In the following days, I plan to allocate time to test it on Tensorflow 2.0.

I have uploaded the new version to my personal page, https://github.com/zouguojian/Traffic-speed-prediction/tree/main/3S-TBLN/baseline/DCRNN. you can download these codes on your local device to run. looking forward your reply

**Comment by liyaguang** [comment]
Thank you for sharing the code. Would it be possible for you to confirm whether your can train **unmodified code** using **Tensorflow 1.x** (not compat model from 2.x)? 

This [code](https://github.com/zouguojian/Traffic-speed-prediction/blob/main/3S-TBLN/baseline/DCRNN/model/dcrnn_cell.py#L6-L14) appears to be designed for Tensorflow 2.x in compat model rather than native 1.x.
Since the DCRNN Cell implementation relies on the contrib library, which is not officially supported by Tensorflow and could potentially lead to backward compatibility issues. Therefore, it is important to ensure that the code on Tensorflow 1.x without modification to reproduce the results.

In the meantime, I will also investigate how to make the code work with Tensorflow 2.x (in compat model). Thanks for your kind understanding.

**Comment by zouguojian** [comment]
> Thank you for sharing the code. Would it be possible for you to confirm whether your can train **unmodified code** using **Tensorflow 1.x** (not compat model from 2.x)?
> 
> This [code](https://github.com/zouguojian/Traffic-speed-prediction/blob/main/3S-TBLN/baseline/DCRNN/model/dcrnn_cell.py#L6-L14) appears to be designed for Tensorflow 2.x in compat model rather than native 1.x. Since the DCRNN Cell implementation relies on the contrib library, which is not officially supported by Tensorflow and could potentially lead to backward compatibility issues. Therefore, it is important to ensure that the code on Tensorflow 1.x without modification to reproduce the results.
> 
> In the meantime, I will also investigate how to make the code work with Tensorflow 2.x (in compat model). Thanks for your kind understanding.

Yep, dear Li, maybe you are right. I have used the high version of TensorFlow to be compatible low version.

**Comment by zouguojian** [comment]
I have solved this problem in PyTorch！！！

**Comment by liyaguang** [comment]
Great job! It's wonderful to hear that you were able to solve the problem using PyTorch! If you want to, feel free to share your solution with others

**Comment by zouguojian** [comment]
> Great job! It's wonderful to hear that you were able to solve the problem using PyTorch! If you want to, feel free to share your solution with others
of cause, I will public it in the next week

**Comment by Gs0098** [comment]
> > Great job! It's wonderful to hear that you were able to solve the problem using PyTorch! If you want to, feel free to share your solution with others干得漂亮！很高兴听到你能够使用 PyTorch 解决这个问题！如果你想的话，随时可以与他人分享你的解决方案！
> > of cause, I will public it in the next week当然，我将在下周公布它

"Hello, esteemed researcher. May I ask where you have published your PyTorch solution?"

**Comment by zouguojian** [comment]
> > > Great job! It's wonderful to hear that you were able to solve the problem using PyTorch! If you want to, feel free to share your solution with others干得漂亮！很高兴听到你能够使用 PyTorch 解决这个问题！如果你想的话，随时可以与他人分享你的解决方案！
> > > of cause, I will public it in the next week当然，我将在下周公布它
> 
> "Hello, esteemed researcher. May I ask where you have published your PyTorch solution?"

is here https://github.com/zouguojian/Traffic-speed-prediction/tree/main/3S-TBLN/baselines/DCRNN
